presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -701,6 +701,12 @@ else if (!found && (Boolean) invokeOperator(OperatorType.EQUAL, types(node.getVa
             if (hasUnresolvedValue) {
                 Type type = type(node.getValue());
                 List<Expression> expressionValues = toExpressions(values, types);
+
+                // transform IN expression with single literal value into an equality comparison
+                if (expressionValues.size() == 1 && expressionValues.get(0) instanceof Literal) {
```

Comment:
Why does this need to be constrained to literals? Wouldn't we want to do it for something like: `x IN (y + 1)` => `x = y + 1` ?